plugin/cmp.lua

```diff
@@ -122,7 +122,11 @@ if vim.on_key then
   end, vim.api.nvim_create_namespace('cmp.plugin'))
 end
 
-vim.cmd [[command! CmpStatus lua require('cmp').status()]]
+vim.api.nvim_create_user_command(
+  'CmpStatus',
+  require('cmp').status,
```

Comment:
Please be aware that it will require cmp at the time that command is created. To avoid that it should be wrapped in lambda/anonymous function.

Reply:
I was thinking this as well. By the time this file is loaded doesn't that imply that `cmp` is available at the time? Is there a case where this `plugin/cmp.lua` file will be loaded but the plugin itself will not?

Reply:
I tested this `nvim_create_user_command` with some pretty heavy lazy loading in different cases and couldn't get it in a state where this caused issues when the `vim.api.nvim_create_user_command` was called.

Reply:
True, but `plugin/cmp.lua` will be sourced by nvim as a startup script (if added to rtp). So those requires are happening any way https://github.com/mehalter/nvim-cmp/blob/command_api/plugin/cmp.lua#L6-L10.

Reply:
I see, so just to clarify, should I update it to wrap it in an anonymous function?

Reply:
@hrsh7th Have you got an opinion here?

Reply:
For my personal taste, I think it's better to wrap it in an anonymous function. (This is because we can eliminate the assumption that the arguments given to the anonymous function implicitly match the status function.)

_google translated_

Reply:
This is a great point @hrsh7th I have updated the commit to wrap it in an anonymous function.